src/shims/io.rs

```diff
@@ -0,0 +1,167 @@
+        }
+
+        let fd = this.read_scalar(fd_op)?.to_i32()?;
+        let cmd = this.read_scalar(cmd_op)?.to_i32()?;
+
+        if cmd == this.eval_libc_i32("F_SETFD")? {
+            // This does not affect the file itself. Certain flags might require changing the file
+            // or the way it is accessed somehow.
+            let flag = this.read_scalar(arg_op.unwrap())?.to_i32()?;
+            // The only usage of this in stdlib at the moment is to enable the `FD_CLOEXEC` flag.
+            let fd_cloexec = this.eval_libc_i32("FD_CLOEXEC")?;
+            if let Some(FileHandle{ flag: old_flag, .. }) = this.machine.file_handler.handles.get_mut(&fd) {
+                if flag ^ *old_flag == fd_cloexec {
+                    *old_flag = flag;
+                } else {
+                    throw_unsup_format!("Unsupported arg {:#x} for `F_SETFD`", flag);
+                }
+            }
+            Ok(0)
+        } else if cmd == this.eval_libc_i32("F_GETFD")? {
+            if let Some(handle) = this.machine.file_handler.handles.get(&fd) {
+                Ok(handle.flag)
+            } else {
+                this.machine.last_error = this.eval_libc_i32("EBADF")? as u32;
+                Ok(-1)
+            }
+        } else {
+            throw_unsup_format!("Unsupported command {:#x}", cmd);
+        }
+    }
+
+    fn close(&mut self, fd_op: OpTy<'tcx, Tag>) -> InterpResult<'tcx, i32> {
+        let this = self.eval_context_mut();
+
+        if !this.machine.communicate {
+            throw_unsup_format!("`open` not available when isolation is enabled")
+        }
+
+        let fd = this.read_scalar(fd_op)?.to_i32()?;
+
+        if let Some(handle) = this.machine.file_handler.handles.remove(&fd) {
+            this.consume_result::<i32>(handle.file.sync_all().map(|_| 0), -1)
+        } else {
+            this.machine.last_error = this.eval_libc_i32("EBADF")? as u32;
+            Ok(-1)
+        }
+    }
+
+    fn read(
+        &mut self,
+        fd_op: OpTy<'tcx, Tag>,
+        buf_op: OpTy<'tcx, Tag>,
+        count_op: OpTy<'tcx, Tag>,
+    ) -> InterpResult<'tcx, i64> {
+        let this = self.eval_context_mut();
+
+        if !this.machine.communicate {
+            throw_unsup_format!("`open` not available when isolation is enabled")
+        }
+
+        let tcx = &{ this.tcx.tcx };
+
+        let fd = this.read_scalar(fd_op)?.to_i32()?;
+        let buf = this.force_ptr(this.read_scalar(buf_op)?.not_undef()?)?;
+        let count = this.read_scalar(count_op)?.to_usize(&*this.tcx)?;
+
+        let mut bytes = vec![0; count as usize];
+
+        let read_result = if let Some(FileHandle { file, ..}) = this.machine.file_handler.handles.get_mut(&fd) {
+            file.read(&mut bytes).map(|bytes| bytes as i64)
+        } else {
+            this.machine.last_error = this.eval_libc_i32("EBADF")? as u32;
+            return Ok(-1);
+        };
+
+        let read_bytes = this.consume_result::<i64>(read_result, -1)?;
```

Comment:
@oli-obk, for example here I could just use `-1i32` and then cast it to an `i64` however I'm not sure if it is ok to just cast `read_result` to an `i32` instead.